api-tests/specs/practitioner-service/linkperson.js

```diff
@@ -0,0 +1,83 @@
+        const firstPersonResponse = await baseRequest.post(practitionerEndpointUri).send(requests.createFirstPerson)
+        const captureFirstPersonResponseData = firstPersonResponse.text
+        let parseFirstPersonResponseData = JSON.parse(captureFirstPersonResponseData)
+        firstPersonId = parseFirstPersonResponseData.id
+        
+        //link person to a user
+        let linkUserJSON = requests.linkUser
+        linkpersonEndpointUri = linkpersonEndpointUri.replace("{personId}", firstPersonId);
+        const response = await baseRequest.post(linkpersonEndpointUri).send(linkUserJSON)
+
+        //check application response confirms link success
+        expect(response.status).to.equal(HttpStatus.OK)
+
+    });
+
+    it.only('Given a Person exists who is not linked to a User When I submit an API request to link the Person to a User who is already linked to a Person in this trial Then the Person and User are NOT linked and I receive an error notification ', async () => {
+
+        //post a request to practitioner end point to create second Practitioner
+        const secondPersonResponse = await baseRequest.post(practitionerEndpointUri).send(requests.createSecondPerson)
+        const captureSecondPersonResponseData = secondPersonResponse.text
+        let parseSecondPersonResponseData = JSON.parse(captureSecondPersonResponseData)
+        secondPersonId = parseSecondPersonResponseData.id
+
+        //link person to a user who is already linked
+        let linkUserJSON = requests.linkUser
+        linkpersonEndpointUri = linkpersonEndpointUri.replace("{personId}", secondPersonId);
+        const response = await baseRequest.post(linkpersonEndpointUri).send(linkUserJSON)
+
+        //check application response confirms link failure
+        expect(response.status).to.equal(422)
+
+    });
+
+    it.only('When I submit an API request with missing Practitioner ID to link a Person who is NOT linked to a User to a User who is not linked to a Person in this trial Then the Person and User are NOT linked and I receive a failure acknowledgement ', async () => {
+
+        //post a request to practitioner end point to create third Practitioner
+        const thirdPersonResponse = await baseRequest.post(practitionerEndpointUri).send(requests.createThirdPerson)
+        const captureThirdPersonResponseData = thirdPersonResponse.text
+        let parseThirdPersonResponseData = JSON.parse(captureThirdPersonResponseData)
+        thirdPersonId = parseThirdPersonResponseData.id
+                
+        //link person to a user with request missing practitioner ID
+        let linkUserJSON = requests.linkUser
+        linkpersonEndpointUri = linkpersonEndpointUri.replace("{personId}", " ");
+        const response = await baseRequest.post(linkpersonEndpointUri).send(linkUserJSON)
+
+        //check application response confirms link failure
+        expect(response.status).to.equal(422)
```

Comment:
Can we use ( HttpStatus.UNPROCESSABLE_ENTITY) instead of the status error code please

Reply:
The suggested changes have been made and pushed to the test branch.